content/blog/improved-refresh-destroy-experience/index.md

```diff
@@ -0,0 +1,237 @@
+---
+title: "Improved refresh and destroy experience"
+date: 2025-04-01
+meta_desc: "Refreshing and destroying programs will now use updates from your program"
+authors:
+    - fraser-waters
+    - meagan-cojocar
+tags:
+    - features
+---
+
+Pulumi enables teams to manage their infrastructure using the programming languages and tools they are already familiar with, supporting use cases such as more complex authentication workflows, dynamically configured resources, and more. Loops, conditionals, environment variables, external calls -- if your language can do it, `pulumi up` will run it.
+
+The `pulumi refresh` and `pulumi destroy` commands sit alongside `pulumi up` and support refreshing and destroying resources in your stack. While `refresh` and `destroy` take into account updated configuration (such as that specified in your `Pulumi.<stack>.yaml`), they have historicall not taken into account changes in your *code*. For stacks which depend on code to update credentials for a provider, or to determine which resources to create, this can break or severely hamper the use of `refresh` and `destroy`. Well, no more! As of the latest release of Pulumi, the `pulumi refresh` and `pulumi destroy` commands now support the `--run-program` flag, which allows you to run your program before refreshing or destroying your stack. (And no, this is not an April Fool's joke!)
```

Comment:
> take into account updated configuration

Not really. The only thing they actually use any environment variables set by ESC.